osf/models/node.py

```diff
@@ -2420,6 +2483,34 @@ def add_addon(self, name, auth, log=True):
             self.save()  # TODO Required?
         return ret
 
+    def has_addon_on_children(self, addon):
+        """Checks if a given node has a specific addon on child nodes
+            that are not registrations or deleted
+        """
+        if self.has_addon(addon):
+            return True
+
+        for node_relation in self.node_relations.filter(is_node_link=False, child__is_deleted=False).select_related('child'):
```

Comment:
This could perform better with a custom sql query, we could do it all in one. Might be worth a `# TODO`